### PR TITLE
Amb 1745 complete search endpoint

### PR DESCRIPTION
**Pull request by artronics**
## Summary
* Routine Change
* :exclamation: Breaking Change
* :robot: Operational or Infrastructure Change
* :sparkles: New Feature
* :warning: Potential issues that might be caused by this change

Add any other relevant notes or explanations here. **Remove this line if you have nothing to add.**

## Reviews Required
* [x] Dev
* [ ] Test
* [ ] Tech Author
* [ ] Product Owner

## Review Checklist
:information_source: This section is to be filled in by the **reviewer**.

* [ ] I have reviewed the changes in this PR and they fill all or part of the acceptance criteria of the ticket, and the code is in a mergeable state.
* [ ] If there were infrastructure, operational, or build changes, I have made sure there is sufficient evidence that the changes will work.
* [ ] I have ensured the changelog has been updated by the submitter, if necessary.

### CODE REVIEW

File: lambda_code/src/fhir_controller.py [hunk 1]
```diff
@@ -42,11 +42,8 @@ def get_immunization_by_id(self, aws_event) -> dict:
     def create_immunization(self, aws_event):
         try:
             imms = json.loads(aws_event["body"])
-        except json.decoder.JSONDecodeError as error:
-            body = create_operation_outcome(resource_id=str(uuid.uuid4()), severity=Severity.error,
-                                            code=Code.invalid,
-                                            diagnostics=f"Request's body contains malformed JSON\n{error}")
-            return self.create_response(400, body.json())
+        except json.decoder.JSONDecodeError:
+            return self._create_bad_request("Request's body contains malformed JSON")
```
Comment (lee-richardson-nhs) [inline lambda_code/src/fhir_controller.py:46]: Should tell the caller what that error is.

File: lambda_code/src/fhir_controller.py [hunk 2]
```diff
@@ -66,6 +63,19 @@ def delete_immunization(self, aws_event):
         except UnhandledResponseError as unhandled_error:
             return self.create_response(500, unhandled_error.to_operation_outcome().json())
 
+    def search_immunizations(self, aws_event) -> dict:
+        params = aws_event["queryStringParameters"]
+        print(aws_event)
```
Comment (lee-richardson-nhs) [inline lambda_code/src/fhir_controller.py:68]: Debug code.

File: lambda_code/src/fhir_controller.py [hunk 3]
```diff
@@ -66,6 +63,19 @@ def delete_immunization(self, aws_event):
...
+
+        if "nhsNumber" not in params:
```
Comment (lee-richardson-nhs) [inline lambda_code/src/fhir_controller.py:70]: Should check that the value isn't None or whitespace.

File: lambda_code/src/fhir_repository.py [hunk 4]
```diff
@@ -68,6 +78,42 @@ def delete_immunization(self, imms_id: str) -> dict:
                 raise UnhandledResponseError(message=f"Unhandled error from dynamodb: {e.response['Error']['Code']}",
                                              response=e.response)
 
+    def find_immunizations(self, nhs_number: str, disease_code: str):
+        """it should find all patient's Immunization events for a specified disease_code"""
+        # TODO:
```
Comment (lee-richardson-nhs) [inline lambda_code/src/fhir_repository.py:83]: All of these need to be done before this is merged.

**Review by lee-richardson-nhs — CHANGES_REQUESTED** [review]

File: lambda_code/src/fhir_controller.py (hunk 3 above)
Comment (artronics) [reply lambda_code/src/fhir_controller.py:70]: This part should be done when validation code is wired in. For now I just assumed the object is valid.

File: lambda_code/src/fhir_repository.py (hunk 4 above)
Comment (artronics) [reply lambda_code/src/fhir_repository.py:83]: No we have some of the answers. There are still things that should be left there. I'll update the comment.

**Review by lee-richardson-nhs — APPROVED** [review]

**Pull request merged**